aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/model/GetObjectRequest.java

```diff
@@ -1008,4 +1008,47 @@ public GetObjectRequest withS3ObjectId(S3ObjectId s3ObjectId) {
         setS3ObjectId(s3ObjectId);
         return this;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) return true;
+        if (object == null || getClass() != object.getClass()) return false;
```

Comment:
instanceof returns true for subtypes

I would personally be surprised if an object of type A compared to an object of type B extends A returned true. Is this the desired behavior?